File: Data structure and algorithms/后缀表达式.py
def suffix(s):
    stack=[]
    for i in s:
        if i!='+' and i!='-' and i!='/' and i!='*' and i!='#':
            stack.append(i)
        elif i=='+':
            a=int(stack.pop())
            if not stack:
                print(f'Expression Error: {a}')
                return 
            b=int(stack.pop())
            stack.append(b+a)
        elif i=='-':
            a=int(stack.pop())
            if not stack:
                print(f'Expression Error: {a}')
                return 
            b=int(stack.pop())
            stack.append(b-a)
        elif i=='*':
            a=int(stack.pop())
            if not stack:
                print(f'Expression Error: {a}')
                return 
            b=int(stack.pop())
            stack.append(b*a)
        elif i=='/':
            a=int(stack.pop())
            if not stack:
                print(f'Expression Error: {a}')
                return 
            b=int(stack.pop())
            if a==0:
                print(f'Error: {b}/{a}')
                return 
            stack.append(b//a)
        elif i=='#':
            c=stack.pop()
            if not stack:
                print(c)
            if stack:
                print(f'Expression Error: {c}')

s=list(map(str,input().split()))
suffix(s)

File: C++/homework/后缀表达式.cpp
#include<iostream>
#include<stack>
#include<string>
using namespace std;
int main(){
    string s;
    getline(cin,s);
    stack <int> stk;
    int top=0;
    int res=0;
    for(int i=0;i<s.length();i++){
        if(s[i]!='+'&&s[i]!='-'&&s[i]!='/'&&s[i]!='*'&&s[i]!='#'){
            stk.push(s[i]-'0');
        }
        else if(i=='+'){
            top=stk.top();
            stk.pop();
            if(stk.empty()){
                cout<<"Expression Error:";
            }
            stk.push(top+stk.top());
            stk.pop();
        }
        else if (i=='-')
        {
            top=stk.top();
            stk.pop();
            res+=top;
        }
        else if(i=='*'){
            top=stk.top();
            stk.pop();
            res*=top;
        }
        else if(i=='/'){
            top=stk.top();
            stk.pop();
            res/=top;
        }
    }
}
  